ch_pipeline/processing/daily.py

```diff
@@ -429,10 +513,90 @@ def _available_tags(self):
             csd_set = set(csds)
             csds += [csd for csd in csd_i if csd not in csd_set]
 
+        # grab the list of files that are online, and that exist anywhere, from the earliest csd to the latest
+        csds_sorted = sorted(csds)
+        filenames_online, filenames_that_exist = self._available_files(
+            csds_sorted[0], csds_sorted[-1] + 1
+        )
+
+        # only queue jobs for which all data is available online in chime_online
+        csds_available = self._csds_available_data(
+            csds_sorted, filenames_online, filenames_that_exist
+        )
+        csds = [csd for csd in csds if csd in csds_available]
+
         tags = ["%i" % csd for csd in csds]
 
         return tags
 
+    def _csds_available_data(self, csds, filenames_online, filenames_that_exist):
+        """
+        Return the subset of csds in `csds` for whom all files are online.
+
+        `filenames_online` and `filenames_that_exist` are a list of namedtuples or dictionaries,
+        with attributes `start_time` and `finish_time`.
+
+        """
+        csds_available = []
+
+        for csd in csds:
+            start_time = ephemeris.csd_to_unix(csd)
+            end_time = ephemeris.csd_to_unix(csd + 1)
+
+            # online - list of filenames that are online between start_time and end_time
+            # index_online, the final index in which data was located
+            online, index_online = self._files_in_timespan(
+                start_time, end_time, filenames_online
+            )
+            exists, index_exists = self._files_in_timespan(
+                start_time, end_time, filenames_that_exist
+            )
+
+            if (len(online) == len(exists)) and (len(online) != 0):
+                csds_available.append(csd)
+
+            # Since the csds, filenames_online, and filenames_that_exist are sorted by time,
+            # and non-overlapping
+            # as we check a csd, we are going to index further into the lists.
+            index_online = max(index_online - 3, 0)
```

Comment:
I still don't get the -3 in here. As you state the files are sorted and guaranteed non-overlapping so I don't think you need to roll the index back at all here.

Reply:
A file can contain more than one sidereal day! But, I guess if we are not interested in the exact number of files, then it does not matter. One file will be dropped from the `online` and `exists` list. Okay, I will remove it.

Reply:
Oh I see. I don't think dropping a file from the lists is okay.

I was confused about what you return from file_in_timespan. I think either you need to either return the last file that was in the span (and currently its the first file outside it), or subtract one (but only one, not three) from the index outside of that function.